### DISCUSSION TITLE
Require 2FA doesn't do anything

### DISCUSSION THREAD

**Discussion by dParikesit** [directus/directus#13679] [Archive]
Hello, can someone explains what "require 2FA" do? 
I've enabled it and configure the system permissions to app access minimum and recommended defaults, the collection permissions are all access on CRUD. 

The problem are users can't enable 2FA (forbidden), and they are able to edit the collection even if they haven't enable 2FA.
Any suggestions?

**Comment by azrikahar** [comment]
> Hello, can someone explains what "require 2FA" do?

Currently it's meant to force 2FA on login. However as you described, it's not properly enforcing the 2FA setup upon login. This issue is being tracked in #5682, and the PR #7805 will resolve this 👍

> I've enabled it and configure the system permissions to app access minimum and recommended defaults, the collection permissions are all access on CRUD.
>
> The problem are users can't enable 2FA (forbidden)

This is definitely an actual issue, and has been reported in directus/directus#9883 (which will be fixed in the same PR!). Tbe recommended defaults were missing the permission for this field, hence the forbidden error. In the meantime, you will need to manually grant update permission to the `tfa_secret` field for the role(s) you wanted to use 2FA 👍

> and they are able to edit the collection even if they haven't enable 2FA.

Currently 2FA doesn't enforce _during_ editing, so this would still be the same/expected behavior even after #7805 is merged.

**Reply by dParikesit** [reply]
Thanks. Is there any estimation on when it will be released?

**Reply by azrikahar** [reply]
The tentative estimate would be 1-2 weeks[^1].

[^1]: Though like everything else, it is subject to change.
